jcl/src/java.management/share/classes/com/ibm/java/lang/management/internal/ThreadMXBeanImpl.java

```diff
@@ -280,34 +280,51 @@ public ThreadInfo getThreadInfo(long id) {
 	 */
 	@Override
 	public ThreadInfo[] getThreadInfo(long[] ids, boolean lockedMonitors,
-			boolean lockedSynchronizers) {
-		// Verify inputs
-		if (lockedMonitors && !isObjectMonitorUsageSupported()) {
-			/*[MSG "K05FC", "Monitoring of object monitors is not supported on this virtual machine"]*/
-			throw new UnsupportedOperationException(com.ibm.oti.util.Msg.getString("K05FC")); //$NON-NLS-1$
-		}
-		if (lockedSynchronizers && !isSynchronizerUsageSupported()) {
-			/*[MSG "K05FB", "Monitoring of ownable synchronizer usage is not supported on this virtual machine"]*/
-			throw new UnsupportedOperationException(com.ibm.oti.util.Msg.getString("K05FB")); //$NON-NLS-1$
-		}
-
-		SecurityManager security = System.getSecurityManager();
-		if (security != null) {
-			security.checkPermission(ManagementPermissionHelper.MPMONITOR);
-		}
-
-		// Validate thread ids
-		for (long id : ids) {
-			if (id <= 0) {
-				/*[MSG "K05F7", "Thread id must be greater than 0."]*/
-				throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K05F7")); //$NON-NLS-1$
-			}
-		}
-
-		// Create an array and populate with individual ThreadInfos
-		return this.getMultiThreadInfoImpl(ids, Integer.MAX_VALUE,
-				lockedMonitors, lockedSynchronizers);
-	}
+            boolean lockedSynchronizers) {
+        return getThreadInfoCommon(ids, lockedMonitors, lockedSynchronizers, Integer.MAX_VALUE);
+    }
+
+	/*[IF Java10]*/
+    @Override
+    public ThreadInfo[] getThreadInfo(long[] ids, boolean lockedMonitors,
+            boolean lockedSynchronizers, int maxDepth) {
+        // K0662 = maxDepth must not be negative.
```

Comment:
These external message declarations are supposed to be preprocessor commands, not comments. Please restore the appropriate preprocessor format.

Reply:
Fixed.